File: src/p0001/cpp/solution.cpp
class Solution {
public:
    static vector<int> twoSum(const vector<int> &nums, int target) {
        unordered_map<int, vector<int>::size_type> dict;

        for (vector<int>::size_type j = 0; j < nums.size(); j++) {
            if (auto i = dict.find(target - nums[j]); i != dict.end()) {
                return {int(i->second), int(j)};
            }

            dict[nums[j]] = j;
        }

        return {};
    }
};

File: src/p0001/golang/solution.go
func twoSum(nums []int, target int) []int {
	dict := map[int]int{}

	for j, y := range nums {
		if i, ok := dict[target-y]; ok {
			return []int{i, j}
		}

		dict[y] = j
	}

	return []int{}
}
